geometric_kernels/eigenfunctions.py

```diff
@@ -171,7 +169,7 @@ def _filter_weights(self, weights: B.Numeric) -> B.Numeric:
         # TODO(VD) write check for this.
         i = 0
         for num in self.num_eigenfunctions_per_level:
-            weights_per_level.append(weights[i] * B.ones(1))
+            weights_per_level.append(weights[i] * B.cast(B.dtype(weights), B.ones(1)))
```

Comment:
You can just use `B.ones(B.dtype(weights), 1)`

Reply:
Done, thanks :)